pinot-segment-local/src/main/java/org/apache/pinot/segment/local/utils/ServerReloadJobStatusCache.java

```diff
@@ -92,4 +106,81 @@ public ReloadJobStatus getOrCreate(String jobId) {
     }
     return status;
   }
+
+  /**
+   * Rebuilds the cache with new configuration and migrates existing entries.
+   * This method is synchronized to prevent concurrent rebuilds.
+   *
+   * @param newConfig new cache configuration to apply
+   */
+  private synchronized void rebuildCache(ServerReloadJobStatusCacheConfig newConfig) {
+    LOG.info("Rebuilding reload status cache with new config: {}", newConfig);
+
+    // Create new cache with new configuration
+    Cache<String, ReloadJobStatus> newCache = CacheBuilder.newBuilder()
+        .maximumSize(newConfig.getMaxSize())
+        .expireAfterWrite(newConfig.getTtlDays(), TimeUnit.DAYS)
+        .recordStats()
+        .build();
+
+    // Migrate existing entries from old cache to new cache
+    Cache<String, ReloadJobStatus> oldCache = _cache;
+    if (oldCache != null) {
+      for (Map.Entry<String, ReloadJobStatus> entry : oldCache.asMap().entrySet()) {
+        newCache.put(entry.getKey(), entry.getValue());
+      }
+    }
+
+    // Atomically swap caches (volatile field ensures safe publication)
+    _cache = newCache;
+    _currentConfig = newConfig;
+
+    LOG.info("Successfully rebuilt reload status cache (size: {})", newCache.size());
+  }
+
+  /**
+   * Maps cluster configuration properties with a common prefix to a config POJO using Jackson.
+   * Uses PinotConfiguration.subset() to extract properties with the given prefix and
+   * Jackson's convertValue() for automatic object mapping.
+   *
+   * @param clusterConfigs map of all cluster configs from ZooKeeper
+   * @param configPrefix prefix to filter configs (e.g., "pinot.server.table.reload.status.cache")
+   * @return ServerReloadJobStatusCacheConfig with values from cluster config, defaults for missing values
+   */
+  @VisibleForTesting
+  static ServerReloadJobStatusCacheConfig buildFromClusterConfig(Map<String, String> clusterConfigs,
+      String configPrefix) {
+    final MapConfiguration mapConfig = new MapConfiguration(clusterConfigs);
+    final PinotConfiguration subsetConfig = new PinotConfiguration(mapConfig).subset(configPrefix);
+    return OBJECT_MAPPER.convertValue(subsetConfig.toMap(), ServerReloadJobStatusCacheConfig.class);
+  }
+
+  /**
+   * Gets the current cache configuration.
+   * Useful for testing and monitoring.
```

Comment:
i see this comment ("Useful for testing and monitoring.") pop up once in a while and it never makes sense IMO to have this as a comment in the code. can we proactively remove them going forward?